### PR TITLE
Added module for managing Apple Mac OSX user defaults

### PR DESCRIPTION
**Pull request by frenck**
##### Issue Type:

New Module Pull Request
##### Ansible Version:

1.7.2
##### Environment:

Apple Mac OS X Yosemite 10.10 (build 14A379b)
##### Summary:

Created a new module for managing Mac's user defaults. With this module you'll be able to tailor most of the settings of your Mac using Ansible. It supports most of defaults datatypes. Array, Boolean, Date, Float, Integer and String types.
##### Steps To Reproduce:

N/A
##### Expected Results:

N/A
##### Actual Results:

N/A

### CODE REVIEW

**Comment by frenck** [comment]
I've just renamed the module from mac_defaults to osx_defaults to be more compatible with the naming of existing modules (e.g. osx_say).

**Comment by josephholsten** [comment]
ha! great minds think alike: #102

File: system/osx_defaults.py [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+    - Apple Mac caches defaults. You may need to logout and login to apply the changes.
+'''
+
+EXAMPLES = '''
+- osx_defaults: domain=com.apple.Safari key=IncludeInternalDebugMenu type=bool value=true state=present
+- osx_defaults: domain=NSGlobalDomain key=AppleMeasurementUnits type=string value=Centimeters state=present
+- osx_defaults: key=AppleMeasurementUnits type=string value=Centimeters
+- osx_defaults:
+    key: AppleLanguages
+    type: array
```
Comment (josephholsten) [inline system/osx_defaults.py:73]: nice!

**Comment by josephholsten** [comment]
Looks like this is much better than my implementation! 

Since you managed to make array-add work, I wonder how you'd tackle:

```
defaults write com.apple.mail NSUserKeyEquivalents -dict-add "Send" -string "@\\U21a9"
```

**Comment by elnappo** [comment]
I'm very interested in this. I hope it gets merged soon. Great work! :)

**Comment by uchida** [comment]
+1, I hope it to be merged.
Would you have a plan to support `-currentHost` switch.
This is a option for machine specific configurations, internal keyboard or trackpad behaviour etc, such as `defaults -currentHost write NSGlobalDomain com.apple.mouse.tapBehavior -int 2`.

**Comment by gregdek** [comment]
Thanks for submitting your module to Ansible Extras. Our policy is that each new module needs to be reviewed and approved by at least two official module reviewers, the list of whom can be found here: https://github.com/ansible/ansible-modules-extras/blob/devel/REVIEWERS.md 

If you know someone on that list, please poke them and encourage them to review your module. :)

To ensure that your module has the best chance of being approved, please double-check that you adhere to the Ansible module guidelines: http://docs.ansible.com/developing_modules.html#module-checklist

**Comment by josephholsten** [comment]
@gregdek thanks for the hint, I'll make sure to do that in future!

**Pull request merged**